pkg/ui/src/views/databases/containers/databaseSummary/index.tsx

```diff
@@ -57,32 +58,40 @@ export class DatabaseSummaryBase extends React.Component<DatabaseSummaryProps, {
   // is a performance concern with invalidation periods.
   loadTableDetails(props = this.props) {
     if (props.tableInfos && props.tableInfos.length > 0) {
-      _.each(props.tableInfos, (tblInfo) => {
-        if (_.isUndefined(tblInfo.numColumns)) {
-          props.refreshTableDetails(new protos.cockroach.server.serverpb.TableDetailsRequest({
-            database: props.name,
-            table: tblInfo.name,
-          }));
-        }
-        if (_.isUndefined(tblInfo.physicalSize)) {
-          props.refreshTableStats(new protos.cockroach.server.serverpb.TableStatsRequest({
-            database: props.name,
-            table: tblInfo.name,
-          }));
-        }
+      _.each(props.tableInfos, (tblInfo, i) => {
```

Comment:
David and I had an IRL conversation about how a more ideal solution (when we endeavor to construct such a thing) might be to lazy load the databases in a way where we wouldn't load an additional one until the previous one was completed. So instead of using a setTimeout cadence (500ms, 1000ms, etc) maybe we could recursively loop through this list instead. Or do something promise-based. Here me taking a swing at it,

```
const loop = (list) => {
  if (!list.length) return; // forgive the one-liner

  const next = list.pop();
  typeof next === "function" && next();
  loop(list);
};

const oneAtATime = props.tableInfos.map(({ numColumns, physicalSize }) => {
  if(_.isUndefined(numColumns)) {
    .... // same
  }
  if (_.isUndefined(physicalSize)) {
  ... // same stuff
  } 
});

loop(oneAtATime);
```

I'm not actually sure what `props.refreshTableDetails` returns, but if it could be a Promise that resolves once the request is completed then the loop function could become `async`,

```
const loop = aysnc (list) => {
  if (!list.length) return; // forgive the one-liner

  const next = list.pop();
  typeof next === "function" && await next();
  loop(list);
};

```

Reply:
@nathanstilwell I ended up using some `async` in the latest update to the code to sequence the calls. Great idea!